ray-operator/test/migration/pre-migration-check.sh

```
@@ -0,0 +1,234 @@
+        
+        echo "--- Head Container Volume Mounts (ray-head) ---"
+        kubectl get raycluster "$name" -n "$NAMESPACE" -o jsonpath='{range .spec.headGroupSpec.template.spec.containers[?(@.name=="ray-head")].volumeMounts[*]}{.name} -> {.mountPath}{"\n"}{end}' 2>/dev/null || echo "(none)"
+        echo ""
+        
+        echo "--- Head Volumes ---"
+        kubectl get raycluster "$name" -n "$NAMESPACE" -o jsonpath='{range .spec.headGroupSpec.template.spec.volumes[*]}{.name}{"\n"}{end}' 2>/dev/null || echo "(none)"
+        echo ""
+        
+        # Worker Groups
+        echo "=== Worker Groups ==="
+        local num_workers=$(kubectl get raycluster "$name" -n "$NAMESPACE" -o jsonpath='{.spec.workerGroupSpecs}' 2>/dev/null | jq 'length' 2>/dev/null || echo "0")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)